test/integration/monitoring-alert/monitoring_alert_test.go

```diff
@@ -0,0 +1,131 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitoring_alert
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/gcloud"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMonitoringAlertSecret(t *testing.T) {
+	email_address := "email@example.com"
+	vars := map[string]interface{}{
+		"email_addresses": []string{email_address},
+	}
+
+	secretT := tft.NewTFBlueprintTest(t, tft.WithVars(vars))
+
+	secretT.DefineVerify(func(assert *assert.Assertions) {
+		secretT.DefaultVerify(assert)
+
+		projectId := secretT.GetStringOutput("project_id")
+		projectDescribe := gcloud.Runf(t, "projects describe %s", projectId)
+		projectNumber := projectDescribe.Get("projectNumber").String()
+		secretDescribe := gcloud.Runf(t, "secrets describe %s --project %s", "secret-1", projectId)
+		secretName := secretDescribe.Get("name").String()
+		assert.Equal(fmt.Sprintf("projects/%s/secrets/secret-1", projectNumber), secretName, "has expected name")
+
+		notificationChannelNames := secretT.GetJsonOutput("notification_channel_names").Array()
+		assert.Len(notificationChannelNames, 1)
+		notificationChannelName := notificationChannelNames[0].String()
+		monitoringChannel := gcloud.Runf(t, "beta monitoring channels list --project %s --filter name='%s'", projectId, notificationChannelName).Array()
+		assert.Len(monitoringChannel, 1)
+		assert.Equal(email_address, monitoringChannel[0].Get("labels.email_address").String())
+
+		monitoringAlerts := gcloud.Runf(t, "alpha monitoring policies list --project %s", projectId).Array()
+		assert.Len(monitoringAlerts, 1)
+		monitoringAlert := monitoringAlerts[0]
+		alertCondition := monitoringAlerts[0].Get("conditions").Array()
+		assert.Len(alertCondition, 1)
+		expectedFilter := fmt.Sprintf("protoPayload.methodName=\"google.cloud.secretmanager.v1.SecretManagerService.DestroySecretVersion\" AND protoPayload.resourceName : \"%s\"", secretT.GetJsonOutput("secret_names").Array()[0].String())
+		assert.Equal(expectedFilter, alertCondition[0].Get("conditionMatchedLog.filter").String())
+		notificationChannel := monitoringAlert.Get("notificationChannels").Array()[0].String()
+		assert.Equal(notificationChannelName, notificationChannel)
+		assert.Equal("WARNING", monitoringAlert.Get("severity").String())
+		assert.Equal("300s", monitoringAlert.Get("alertStrategy.notificationRateLimit.period").String())
+		assert.True(monitoringAlert.Get("enabled").Bool())
+
+		time.Sleep(5 * time.Minute)
+		gcloud.Runf(t, "secrets versions destroy %s/versions/1 -q", secretName)
```

Comment:
would it be useful to test if alerts trigger with different versions are destroyed. for example, if secret has 3 versions, what happens if 1 and 3 are destroyed or just 2

Reply:
In this example, only one secret version is created, so we'd have to modify the example to create more secret versions, which may not be the best for our example... Also, the monitoring alert platform has a minimum notification rate limit of 5 minutes, so if we delete two secret versions in less than 5 minutes, only one notification is triggered.